tasks/docker.py

```diff
@@ -139,10 +140,26 @@ def publish(ctx, src, dst, signed_pull=False, signed_push=False):
     push_env = {}
     if signed_push:
         push_env["DOCKER_CONTENT_TRUST"] = "1"
-    ctx.run(
-        "docker push {dst}".format(dst=dst),
-        env=push_env
-    )
+    remaining_retries = 5
+    while True:
+        warn = True
+        if remaining_retries == 0:
+            warn = False
```

Comment:
Isn't the warn logic swapped? I guess you only want it to complain if it fails after the 5 retries

Reply:
I think it’s in the right order because `warn = True` means “only print a warning instead of raising an exception”:
http://docs.pyinvoke.org/en/stable/api/runners.html#invoke.runners.Runner.run
```
warn (bool) –
Whether to warn and continue, instead of raising
```

For the first tries, in case of failure, I want `invoke` to display a warning message and proceed to the next run of the loop  (`warn = True`) whereas for the last try in the loop, I want the default behavior (`warn = False`) which is raising an `UnexpectedExit` so that the `invoke` command aborts and exit with an error code.

Reply:
Ok... I misunderstood the meaning of `warn`. Thanks for clarifying 🙇‍♀